lib/queuePopulator/QueuePopulator.js

```diff
@@ -74,7 +78,8 @@ class QueuePopulator {
         async.series([
             next => this._setupMetricsClients(next),
             next => this._setupFailedCRRClients(next),
-            next => this._setupZookeeper(() => {
+            next => this._setupZookeeper(next),
+            next => this._setupNotificationConfigManager(() => {
```

Comment:
So finally the notification relies on the queuePopulator ? isnt'it what was discussed here https://github.com/scality/citadel/blob/development/1.0/docs/design/bucket-notification.md#alternatives ?
The inconvenient of this design is that the bucket notification queue populator will live in the CRR queue populator pod, and thus any change of the notification pod will restart the CRR (or vice versa).
Is it finally better to do so or am I missing something ?

Reply:
Yes the current implementation of notifications relies on `queuePopulator` to read entries from log and yes `notification` is still an extension, but new containers are created to host only notification processes. The CRR containers will not have the extension `notification` and the bucket notification containers will only host `notification` processes. This was implemented after internal discussions with @jonathan-gramain.

Reply:
Ref, https://github.com/scality/Federation/commit/351b022151b52a73f5a6ac176031f5f767c69c44

Reply:
so is there a risk calling `_setupFailedCRRClients()` here ?

Reply:
It will be called yes, but the setup will be made only when replication extension is available, https://github.com/scality/backbeat/blob/f17a1821b2604aed59e1ebcbe79336b5c0dce95c/lib/queuePopulator/QueuePopulator.js#L119